server/repository/src/db_diesel/storage_connection.rs

```diff
@@ -19,16 +17,60 @@ const BEGIN_TRANSACTION_STATEMENT: &str = "BEGIN IMMEDIATE;";
 #[cfg(feature = "postgres")]
 const BEGIN_TRANSACTION_STATEMENT: &str = "BEGIN";
 
-pub struct StorageConnection {
-    pub connection: DBConnection,
-    // pub arc_connection: Arc<DBConnection>,
+struct InnerStorageConnection {
+    connection: DBConnection,
+}
+
+/// Helper class to avoid deref_mut() calls, which would require to import DerefMut everywhere we
+/// want to use a connection.
+/// For example, with out it it would look like:
+/// let con: &mut DBConnection = LockedConnection.inner.lock().unwrap().deref_mut();
+pub struct LockedConnection<'a> {
+    inner: MutexGuard<'a, InnerStorageConnection>,
+}
+
+impl<'a> LockedConnection<'a> {
+    pub fn connection(&mut self) -> &mut DBConnection {
+        &mut self.inner.connection
+    }
+
     /// Current level of nested transaction.
     /// For example:
     /// 0 => no transaction
     /// 1 => in transaction
     /// 2 => 1st nested transaction
     /// 3 => 2nd nested transaction
-    pub transaction_level: Cell<i32>,
+    pub fn transaction_level<E>(&mut self) -> Result<i32, TransactionError<E>> {
```

Comment:
Don't know if it is new but there was is a way to get the transaction level from the diesel transaction, using this now...